src/gpu/sycl/prelu_kernels.hpp

```diff
@@ -443,9 +437,13 @@ struct prelu_bwd_kernel_vec_t {
         size_t ithr = item.get_group(0) * conf_.wg_size + item.get_local_id();
         dims_t dims_d, dims_w;
         for (int i = 0; i < max_supported_ndims; i++) {
-            dims_d[i] = (data_md().dims()[i] != 0) ? data_md().dims()[i] : 1;
-            dims_w[i] = (weights_md().dims()[i] != 0) ? weights_md().dims()[i]
-                                                      : 1;
+            dims_d[i] = (data_md().dims()[i] != 0 && i < data_md().ndims())
+                    ? data_md().dims()[i]
+                    : 1;
+            dims_w[i]
+                    = (weights_md().dims()[i] != 0 && i < weights_md().ndims())
+                    ? weights_md().dims()[i]
+                    : 1;
```

Comment:
Please introduce `src_ndims, wei_ndims, src_dim_i and wei_dim_i` for readability purpose.

I'd also replace a `!= 0` check with `> 0` also for readability purpose. Thanks.

Reply:
Done.